File: Java/APISampleCode.java
package apisamplecode;

import java.util.Date;
import java.util.HashMap;
import java.util.Iterator;
import java.util.Map;
import java.util.Set;
/**
 *
 * @author franclin
 */
public class APISampleCode {

  /**
   * @param args the command line arguments
   */
  public static void main(String[] args) {   

    NScAPIWrapper objAPI = new NScAPIWrapper("cdb78aa013683839d85f14adab52022f", "50c229e511ab0", "franclin20.internal.nitrosell.com");
    HashMap products = new HashMap();

    products.put("JWW003", 1);
    products.put("JWW002", 2);
    products.put("JWW001", 1);

    //objAPI.getCustomerDetails("annabel@saintbelle.fr");

    String productStr = "";

    Set set = products.entrySet();
    Iterator iter = set.iterator();

    while (iter.hasNext()) {
      Map.Entry e = (Map.Entry) iter.next();
      productStr += e.getKey() + ":" + e.getValue() + ",";
    }


    HashMap parameters = new HashMap();
    parameters.put("email", "franclin.foping@nitrosell.net");
    parameters.put("ship-firstname", "Jane");
    parameters.put("ship-lastname", "Jane");
    parameters.put("ship-fullname", "Jane Brook");
    parameters.put("ship-company", "");
    parameters.put("ship-address1", "36 Bridge St Row");
    parameters.put("ship-address2", "");
    parameters.put("ship-city", "Chester");
    parameters.put("ship-zippostcode", "UK");
    parameters.put("ship-state", "Cheshire");
    parameters.put("ship-statecode", "UK");
    parameters.put("ship-country", "United Kingdom");
    parameters.put("ship-telephone", "01244400318");
    parameters.put("ship-fax", "01789204015");
    parameters.put("ship-email","franclin.foping@nitrosell.net");
    parameters.put("products", productStr.substring(0, productStr.length() - 1));
    parameters.put("time", new Date().getTime() / 1000);
    parameters.put("userid", objAPI.getUserID());
    
    /*
     * uncomment one these lines to test different situations
    objAPI.getShippingOptions(parameters);    
    objAPI.getShippingAndTax(parameters);
    objAPI.getTaxForAnOrder(parameters);
    */ 
    
    HashMap orderDetails = new HashMap();
    orderDetails.put("email", "franclin.foping@nitrosell.net");
    orderDetails.put("cust-username", "tester");
    orderDetails.put("cust-lastname", "Testing");
    orderDetails.put("cust-address1", "10 White Friars");
    orderDetails.put("cust-city", "Manchester");
    orderDetails.put("cust-country", "United Kingdom");
    orderDetails.put("cust-zipppostcode", "M1 3EU");
    orderDetails.put("ship-firstname", "Jane");
    orderDetails.put("ship-lastname", "Jane");
    orderDetails.put("ship-fullname", "Jane Brook");
    orderDetails.put("ship-company", "");
    orderDetails.put("ship-address1", "36 Bridge St Row");
    orderDetails.put("ship-address2", "");
    orderDetails.put("ship-city", "Chester");
    orderDetails.put("ship-zippostcode", "UK");
    orderDetails.put("ship-state", "Cheshire");
    orderDetails.put("ship-statecode", "UK");
    orderDetails.put("ship-country", "United Kingdom");
    orderDetails.put("ship-telephone", "01244400318");
    orderDetails.put("ship-fax", "01789204015");
    orderDetails.put("ship-email","franclin.foping@nitrosell.net");
    orderDetails.put("products", productStr.substring(0, productStr.length() - 1));
    orderDetails.put("time", new Date().getTime() / 1000);    
    orderDetails.put("shippingid", "163");
    orderDetails.put("tenderid", "10");
    orderDetails.put("userid", objAPI.getUserID());
    
    objAPI.insertWebOrders(orderDetails);
  }
}

File: PHP/APISampleCode.php
<?php
/**
 *  This file contains some sample code.
 *  It describes how you can interact with the API
 */
require_once 'NScAPIWrapper.php';

$objAPI = new NScAPIWrapper("YOUR PRIVATE KEY", "YOUR USER ID", "YOUR WEBSTORE URL", "YOUR SYNC USERNAME", "YOUR SYNC PASSWORD");


echo $objAPI->getCustomerDetails("you@domain.com").PHP_EOL;

exit;
$arrParameters = array(
    'email' => 'you@yourself.com',
    'ship-firstname' => 'Jane',
    'ship-lastname' => 'Jane',
    'ship-fullname' => 'Jane Brook',
    'ship-company' => '',
    'ship-address1' => '36 Bridge St Row',
    'ship-address2' => '',
    'ship-city' => 'Chester',
    'ship-zippostcode' => 'CH1 1NN',
    'ship-state' => 'Cheshire',
    'ship-statecode' => 'UK',
    'ship-country' => 'United Kingdom',
    'ship-countrycode' => 'UK',
    'ship-telephone' => '01244400318',
    'ship-fax' => '01789204015',
    'ship-email' => 'you@yourself.net',
    'products' => 'JWW003:3,JWW002:1,JWC001:1',
    'time' => time(),
    'userid' => $objAPI->getUserID()
);


// Uncomment one of the following lines to test different functions.

//$getShippingOptionsArgs = "email=franclin.foping@nitrosell.net&ship-firstname=Jane&ship-lastname=Jane&ship-fullname=Jane+Brooks&ship-company=&ship-address1=36+Bridge+St+Row&ship-address2=&ship-city=Chester&ship-zippostcode=CH1+1NN&ship-state=Cheshire&ship-statecode=UK&ship-country=United+Kingdom&ship-countrycode=UK&ship-telephone=01244400318&ship-fax=01789204015&ship-email=franclin.foping%40nitrosell.net&products=JWW003%3A3%2CJWW002%3A1%2CJWC001%3A1&time=".time()."&userid=".$objAPI->getUserID();

//echo $objAPI->getShippingOptions($getShippingOptionsArgs).PHP_EOL;

//echo $objAPI->getShippingAndTax($getShippingOptionsArgs).PHP_EOL;

//echo $objAPI->getShippingAndTax(http_build_query($arrParameters)).PHP_EOL;


//$orderParameters = "email=franclin.foping@nitrosell.net&cust-username=tester&cust-firstname=Test&cust-lastname=Testing&cust-address1=10+White+Friars&cust-city=Manchester&cust-country=United+Kingdom&cust-zippostcode=M1+3EU&ship-firstname=Jane&ship-lastname=Jane&ship-fullname=Jane+Brooks&ship-company=&ship-address1=36+Bridge+St+Row&ship-address2=&ship-city=Chester&ship-zippostcode=CH1+1NN&ship-state=Cheshire&ship-statecode=UK&ship-country=United+Kingdom&ship-countrycode=UK&ship-telephone=01244400318&ship-fax=01789204015&ship-email=franclin.foping%40nitrosell.net&products=JWW003%3A3%2CJWW002%3A1%2CJWC001%3A1&time=".  time()."&shippingid=163&tenderid=10&userid=".$objAPI->getUserID();

$arrOrderParameters = array(
    'email' => 'you@nitrosell.com',
    'cust-username' => 'tester',
    'cust-lastname' => 'Testing',
    'cust-address1' => '10 White Friars',
    'cust-city' => 'Manchester',
    'cust-country' => 'United Kingdom',
    'cust-zippostcode' => 'M1 3EU',
    'ship-firstname' => 'Jane',
    'ship-lastname' => 'Jane',
    'ship-fullname' => 'Jane Brook',
    'ship-company' => '',
    'ship-address1' => '36 Bridge St Row',
    'ship-address2' => '',
    'ship-city' => 'Chester',
    'ship-zippostcode' => 'CH1 1NN',
    'ship-state' => 'Cheshire',
    'ship-statecode' => 'UK',
    'ship-country' => 'United Kingdom',
    'ship-countrycode' => 'UK',
    'ship-telephone' => '01244400318',
    'ship-fax' => '01789204015',
    'ship-email' => 'you@nitrosell.com',
    'products' => 'JWW003:3,JWW002:1,JWC001:1',
    'time' => time(),
    'shippingid' => '163',
    'tenderid' => '10',
    'userid' => $objAPI->getUserID()
);

//echo $objAPI->insertWebOrders(http_build_query($arrOrderParameters)).PHP_EOL;

// We update a given product with new stock, code, name, and price.
$arrOrderParameters = array(
  array(
    "product_id" => 1, // please ensure that this product ID corresponds to the actual product to be updated
    "product_stock" => 100,
    "product_code" => "New SKU",
    "product_name" => "New prod name",
    "product_price" => 10,
  )
);

// uncomment the line below to test products updating
//echo $objAPI->updateProducts(http_build_query($arrOrderParameters)).PHP_EOL;

?>
